app/modules/intelligence/tracing/logfire_tracer.py

```
@@ -147,6 +170,123 @@ def is_logfire_enabled() -> bool:
+        import logfire
+
+        with logfire.set_baggage(**attrs):
+            span_attrs = {"environment": attrs["environment"]}
+            if attrs.get("user_id"):
+                span_attrs["user_id"] = attrs["user_id"]
+            for k, v in attrs.items():
+                if k not in ("user_id", "environment") and v:
+                    span_attrs[k] = v
+            with logfire.span("llm_call", **span_attrs):
+                yield
+    except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)